### PR TITLE
Input events

### PR DESCRIPTION
**Pull request by marcandre**
While not widely known, the `input` event is triggered before all `change` events for inputs and selects (see [the spec here](https://www.w3.org/TR/html5/forms.html#send-select-update-notifications))

This PR fixes this. Also adds a blurb on how to run tests...

I haven't changed the prototype version because it uses an old library that does not support the `input` event. Last commit was 4 years ago, looks like abandonware.

FWIW, `Parsley` relies on the `input` event, so the current code is [causing bad interactions](https://github.com/guillaumepotier/Parsley.js/issues/1134).

Thanks

### CODE REVIEW

**Comment by marcandre** [comment]
PS: If you really want, it's easy to update the library, but then it has to be stated that only the one you specify has to be used...

File: coffee/chosen.jquery.coffee [hunk 1]
```diff
@@ -510,3 +510,7 @@ class Chosen extends AbstractChosen
         w = f_width - 10
 
       @search_field.css({'width': w + 'px'})
+
+  form_field_has_changed: (extra) ->
```
Comment (koenpunt) [inline coffee/chosen.jquery.coffee]: Also I'm not sure about the name of the method. More leaning to `trigger_form_field_change`, but maybe some of @harvesthq/chosen-developers can weigh in on this matter?

File: contributing.md [hunk 2]
```diff
@@ -188,3 +188,13 @@ Once you're configured, building the JavaScript from the command line is easy:
     grunt watch                # watch coffee/ for changes and build Chosen
 
 If you're interested, you can find the task in [Gruntfile.coffee](https://github.com/harvesthq/chosen/blob/master/Gruntfile.coffee).
+
+#### Running tests
+
+To install all development dependencies, in the project's root directory, run
```
Comment (marcandre) [inline contributing.md]: It is indeed. As an experienced dev, the *only* thing I'm interested in is how do I get to run the tests. For this reason, I thought a separate (and independant) section was better. I probably chose the wrong header size. Let me tweak it and split this into another PR.

File: coffee/chosen.jquery.coffee (hunk 1 above)
Comment (marcandre) [inline coffee/chosen.jquery.coffee]: `trigger_form_field_change` was my first idea :smile:, I thought it might be best to be a bit less specific in the name, but I'm fine with that name too.
You don't use `_myPrivateMethod` for private/protected methods, right? It's not really a public method.

**Comment by marcandre** [comment]
OK, PR updated, with factorization done for prototype's version too.

**Comment by koenpunt** [comment]
Looks good, thanks! 
@pfiller @stof @tjschuck second opinions?

**Comment by koenpunt** [comment]
@marcandre would it be easily possible to feature-test the prototype version to see if it has support for `input`?

**Comment by marcandre** [comment]
The only way would be to know is to try to call `simulate('input')` and rescue an exception...

The lib isn't all that useful anyways... Want a PR to get rid of it?

**Comment by marcandre** [comment]
PR created, getting rid of `simulate` altogether.

**Comment by tjschuck** [comment]
@koenpunt This looks good to me — if you're 👍 , I'm 👍

**Comment by tjschuck** [comment]
@marcandre Do you mind squashing this down to one commit?  [Instructions if needed](http://blog.steveklabnik.com/posts/2012-11-08-how-to-squash-commits-in-a-github-pull-request).

**Comment by marcandre** [comment]
Closing in favor of #2704
